### DISCUSSION TITLE
Kiali integration with Zipkin

### DISCUSSION THREAD

**Discussion by ishitharaj** [kiali/kiali#7059] [Q&A]
Hello, I am new to Kiali space and trying to do a project with Kiali and Zipkin. My target is to integrate Kiali with zipkin. I think that I was able to do it partially by using a zipkin http client and  addressing the zipkin API. First I make zipkin api call and then convert data from zipkin response to jaeger model since part of the Kiali code depends on Jaeger models. But Jaeger tracing graph doesn't equal to Zipkin enabled graph in Kiali UI. Maybe some data is missing in when I convert zipkin response to Jaeger model. What is the proper way to achieve this? Appreciate your valuable opinion. 

Moreover, I think that GRPC part will not work since I couldn't find grpc query service in zipkin API.

**Comment by jshaughn** [comment]
Hi @ishitharaj .  Kiali is mainly a consumer of trace information and doesn't really care about the mechanism used to store the traces.  Zipkin can be used to populate traces into Jaeger, as described in Istio documentation.  Perhaps I don't quite understand what you are trying to do with your integration.

cc @josunect

**Comment by josunect** [comment]
Kiali is a consumer of the traces, and the tracing systems have different APIs and might return the data in different formats (Jaeger, OTel, Zipkin). At the moment, Kiali has native support to Jaeger and Tempo. 
@ishitharaj, how's the graph looking at the moment? Probably, some other data needs to be transformed or added. 
The most important data is: 
- The duration: It is calculated with the avg duration of the spans. 
- The list of spans for each trace. 
- The trace ID 

Other attributes used to calculate the color of the traces, traffic map, etc, are: 

- status
- service_name
- node_id
- component
- upstream_cluster
- http.method
- response_flags

Here https://github.com/kiali/kiali/blob/master/tracing/otel/model/converter/converter.go you can see a converter from OTel to Jaeger.
